src/components/login-required.tsx

```diff
@@ -0,0 +1,21 @@
+import React, {FunctionComponent} from 'react'
+import {useRouter} from 'next/router'
+
+export type LoginRequiredParams = {
+  loginRequired: boolean
+}
+
+const LoginRequired: FunctionComponent<LoginRequiredParams> = ({
```

Comment:
this redirects to login if the param is `true`

not sure how robust it is, but it works for now.

Reply:
mostly it just prevents the user from seeing a broken state so if they haXx0r it's on them